build.go

```diff
@@ -12,7 +12,8 @@ import (
 var taskRegistry = task.NewRegistry(task.WithAutoNamespaces(true))
 
 func init() {
-	taskRegistry.Declare("build").Description("build the tools").OptionalArgs("tools").Do(buildscript.BuildTools)
+	taskRegistry.Declare("sa:modtidy").Description("runs go mod tidy").Do(buildscript.SAModTidy)
+	taskRegistry.Declare("build").Description("build the tools").DependsOn("sa:modtidy").OptionalArgs("tools").Do(buildscript.BuildTools)
```

Comment:
instead of making the `build` task depend on this one, let's instead run it in a new `modtidy` task in the Static Analysis buildvariant